lib/Parse/ParsePattern.cpp

```diff
@@ -251,6 +247,13 @@ Parser::parseParameterClause(SourceLoc &leftParenLoc,
     }
 
     if (startsParameterName(*this, isClosure)) {
+      // If let or var is being used as an argument label, allow it but
+      // generate a warning.
+      if (Tok.isAny(tok::kw_let, tok::kw_var)) {
+        diagnose(Tok, diag::parameter_let_var_as_attr, Tok.getText())
+          .fixItReplace(Tok.getLoc(), "`" + Tok.getText().str() + "`");
```

Comment:
Just a general comment - you can run `@swift-ci build toolchain`, download the result and try in Xcode to make sure (I'm not saying that there shouldn't add a new test too) that everything works fine.